kwc-challenge-set.html

```diff
@@ -8,27 +12,469 @@
+                 */
+                progress: {
+                    type: Number,
+                    value: 0
+                },
+                _nextChallenge: {
+                    type: Object,
+                    value: () => {
+                        return {};
+                    }
+                },
+                _numNodes: {
+                    type: Number,
+                    computed: '_computeNumNodes(challengeSet)'
+                },
+                /*
+                 * The radius of each challenge progress circle in the progress bar.
+                 */
+                radius: {
+                    type: Number,
+                    value: 125
+                },
+                /*
+                 * The url of the current user's avatar. If none given a default icon is used.
+                 */
+                avatarUrl: {
+                    type: String,
+                    value: null
+                },
+                _avatarUrl: {
+                    type: String,
+                    computed: '_computeIconUrl("avatar", avatarUrl)'
+                },
+                /*
+                 * A url to use for completed challenges in the progess bar
+                 */
+                completeIconUrl: {
+                    type: String,
+                    value: null
+                },
+                _completeIconUrl: {
+                    type: String,
+                    computed: '_computeIconUrl("complete", completeIconUrl)'
+                },
+                /*
+                 * A url for an icon to use for the current challenge in the progress bar.
+                 */
+                currentIconUrl: {
+                    type: String,
+                    value: null
+                },
+                _currentIconUrl: {
+                    type: String,
+                    computed: '_computeIconUrl("current", currentIconUrl)'
+                },
+                /*
+                 * A url for an icon to used for locked challenges on the progress bar.
+                 */
+                lockedIconUrl: {
+                    type: String,
+                    value: null
+                },
+                _lockedIconUrl: {
+                    type: String,
+                    computed: '_computeIconUrl("locked", lockedIconUrl)'
+                },
+                _completed: {
+                    type: Boolean,
+                    computed: '_computeComplete(progress, challengeSet.challenges.length)'
+                }
+            },
+            observers:  [
+                '_onProgressChanged(progress, radius, challengeSet.*)'
+            ],
+            attached () {
+                this._drawChallengeProgress();
+            },
+            _hasData (challengeSet) {
+                return challengeSet && challengeSet.challenges && challengeSet.challenges.length;
+            },
+            _computeIconUrl (name, propUrl) {
```

Comment:
Outch! Would you mind to store those image data on a separate file? Just so it's easy to browse this file D: